### PR TITLE
[OAP-1848][SQL Data Source Cache] Decouple OneApplicationResource.scala

### PR DESCRIPTION
**Pull request by haojinIntel**
Decouple OneApplicationResource.scala and create new REST API to get cacheStates.

### CODE REVIEW

**Comment by haojinIntel** [comment]
@LuciferYang @zhixingheyi-tian Please help to review this PR. Thanks.

**Comment by LuciferYang** [comment]
Good work @haojinIntel ~ But can u add a runtime UI snapshot of new OapTab?  I think UT not cover this

File: oap-cache/oap/src/main/scala/org/apache/spark/sql/hive/thriftserver/OapEnv.scala [hunk 1]
```diff
@@ -45,7 +42,6 @@ private[spark] object OapEnv extends Logging {
       sc.addSparkListener(new OapListener)
       this.sparkSession = SparkSession.getActiveSession.get
       sc.ui.foreach(new OapTab(_))
-      initialized = true
     }
```
Comment (zhixingheyi-tian) [inline oap-cache/oap/src/main/scala/org/apache/spark/sql/hive/thriftserver/OapEnv.scala:45]: Ensure  no diff OapEnv.Scala.

Comment (haojinIntel) [reply oap-cache/oap/src/main/scala/org/apache/spark/sql/hive/thriftserver/OapEnv.scala]: Recovered the code.

**Comment by haojinIntel** [comment]
@LuciferYang  Have added the snapshot of new OapTab.
image

File: oap-cache/oap/src/main/scala/org/apache/spark/status/api/v1/OapOneApplicationResource.scala [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  @Path("fibercachemanagers")
+  def fiberList(): Seq[FiberCacheManagerSummary] =
+  {
+    val seqExecutorSummary: Seq[ExecutorSummary] = withUI(_.store.executorList(true))
+    seqExecutorSummary.map(
+      executorSummary =>
+      {
+        val cacheStats = OapRuntime.getOrCreate.fiberSensor.getExecutorToCacheManager.
+          getOrDefault(executorSummary.id, CacheStats())
+        val indexDataCacheSeparationEnable = SparkEnv.get.conf.getBoolean(
```
Comment (LuciferYang) [inline oap-cache/oap/src/main/scala/org/apache/spark/status/api/v1/OapOneApplicationResource.scala:50]: Can we use `get[T](entry: ConfigEntry[T]): T` API, looks simpler

File: oap-cache/oap/src/main/scala/org/apache/spark/status/api/v1/OapOneApplicationResource.scala [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  }
+
+}
+
+private[v1] class OapOneApplicationResource extends OapAbstractApplicationResource {
+
+  @GET
+  def getApp(): ApplicationInfo = {
+    val app = uiRoot.getApplicationInfo(appId)
+    app.getOrElse(throw new NotFoundException("unknown app: " + appId))
```
Comment (LuciferYang) [inline oap-cache/oap/src/main/scala/org/apache/spark/status/api/v1/OapOneApplicationResource.scala:108]: tend to use `s"unknown app: $appId"`

File: oap-cache/oap/src/main/scala/org/apache/spark/status/api/v1/OapOneApplicationResource.scala [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        val cacheStats = OapRuntime.getOrCreate.fiberSensor.getExecutorToCacheManager.
+          getOrDefault(executorSummary.id, CacheStats())
+        val indexDataCacheSeparationEnable = SparkEnv.get.conf.getBoolean(
+          OapConf.OAP_INDEX_DATA_SEPARATION_ENABLE.key,
+          OapConf.OAP_INDEX_DATA_SEPARATION_ENABLE.defaultValue.get) ||
+          SparkEnv.get.conf.getBoolean(
+            OapConf.OAP_INDEX_DATA_SEPARATION_ENABLED.key,
+            OapConf.OAP_INDEX_DATA_SEPARATION_ENABLED.defaultValue.get)
+
+        new FiberCacheManagerSummary(
```
Comment (LuciferYang) [inline oap-cache/oap/src/main/scala/org/apache/spark/status/api/v1/OapOneApplicationResource.scala:57]: I think `FiberCacheManagerSummary` should add a new apply method with `executorSummary`, `cacheStats`, `indexDataCacheSeparationEnable`, the constructor we used looks too complex.

File: oap-cache/oap/src/main/scala/org/apache/spark/status/api/v1/OapOneApplicationResource.scala (hunk 2 above)
Comment (haojinIntel) [reply oap-cache/oap/src/main/scala/org/apache/spark/status/api/v1/OapOneApplicationResource.scala]: Have modified it. Thanks.

**Review by LuciferYang — APPROVED** [review]
LGTM

**Pull request merged**